src/neo4j-arc/graph-visualization/GraphVisualizer/Graph/visualization/Visualization.ts

```diff
@@ -234,6 +234,10 @@ export class Visualization {
     )
 
     this.forceSimulation.updateRelationships(this.graph)
+    // The onGraphChange handler does only repaint relationship color
+    // not width and caption. We work around that by doing an aditional full
+    // render to get the new stylings
```

Comment:
```suggestion
   // The onGraphChange handler does only repaint relationship color
    // not width and caption, since it requires taking into account surrounding data
    // since the arrows have different bending depending on how the nodes are 
    // connected. We work around that by doing an additional full render to get the 
// new stylings
```